### ISSUE TITLE
Plan and timeline for completing volume

### ISSUE THREAD

**Issue by lwjohnst86** [merely-useful/r-rse#1]
Main pedagogical focus is entirely around exercises. They are "first-class" content and are the main thread of the narrative, all other content revolves around them.

Two packages/projects are included in the book:

1. core/example package (Zipf's law) used in explanatory narrative and for (formative) core exercises throughout chapters
2. final/assignment package (Kenya weather data) used in final (summative) exercises in each chapter 

This is the general plan:

- [x] 1. Completely agree on, decide, and finalize the outline and course design in `course-design.md`
- [x] 2. Decide on and complete the description of the core and assignment package
- [x] 3. Decide on and complete the assignment package exercise(s) for each chapter 
- [x] 4. In tandem with 5 below, decide on and write out exact code used or actions taken on the core package in each chapter (including Git commits and other text edits to files).
    - This will be used as the basis for completing 6, since we incorporate them as R code to run automatically (e.g. with git2r, fs package, etc) 
    - For tasks not including code, list actions that will be taken on example package (e.g. CONTRIBUTING guidelines, GitHub labels, etc).
    - Add some learning objectives for the chapter related to the steps taken, the core exercises, and the final exercise.
- [x] 5. Decide on and complete the core package exercises for each chapter that build up and train components for being able to complete the chapter's final project exercise.
    - We can have additional exercises to test or reinforce certain concepts (not related to core or assignment package)
    - Estimated number of exercises might be ~2-4 per in class hour (can have more as course progresses to give more autonomy). Each chapter might be ~3-4 hrs. So total exercises (core + assignment) could be between 6-12. Depends on content though.
- [ ] 6. Build up core package to use within each chapter (more for us to understand where the lesson is at any one point)
    - Can do in tandem with 4, 5, and 7, though this should take less time.
    - How the core package and example packages look like are depends on the steps taken in 4 and the exercises given in 5. They are all entwined, so should be developed in tandem.
    - Idea: Create new repo of this package, with tags/branches as indicators of the state at end of a chapter?
    - Have R do this mostly automatically, so exact output of commits etc can be inserted directly into R Markdown output
- [ ] 7. Once exercises are done, we fill in content around the exercises, modifying from Python RSE as much as we can
- [ ] 8. Decide on and write out instructions to instructors for how to use the course material

Milestones for completing each item would be:

1. November 2020
2. January 2020 (~2 month to complete)
3. Completed in tandem with 2.
4. May 2021 (~4 months to complete)
5. Tied to 4 above.
6. ~June 2021 (\~1 months to complete)~ (*Updated*: Tied to 7.)
7. ~October 2021 (\~4 months to complete)~ (*Updated*) Maybe things delayed us. Aiming for end of 2023
8. ~November 2021 (\~1 month to complete)~ (*Updated*) About a month or two after 7.

After milestone 3, 5, and 7, two or three people should do a start-to-finish review. At the very end, everyone should do a full review of the material, which we will aim to complete within a month.

**Comment by cwickham** [comment]
@lwjohnst86 Just a clarifying question: how do the final project and example package relate? In my head, I have us using Zipf's as the example we build, but the learners are practicing in the exercises on some separate package (the final project).  

So, I see 3. above (the final exercise(s) for each chapter) being steps the learners take to build this final project package.  But, 4. above (the individual exercises) could be using either Zipf's or the final project package.  Is that in line with what you are imagining?

**Comment by lwjohnst86** [comment]
@cwickham sorry for the very delayed response.

I guess it depends on how its structured. When I wrote this I was thinking the final lesson exercise would be like a big "bringing things together" exercise. But actually, as you say, it would be nice to have that final exercise as something to use as a milestone or time for them to work on the final project. And for the individual exercises, yea, I agree, it can be either Zipfs or the final project. I'll revise the language.
